functests/0_config/config.go

```diff
@@ -38,26 +40,41 @@ var _ = Describe("[performance][config] Performance configuration", func() {
 	It("Should successfully deploy the performance profile", func() {
 
 		performanceProfile := testProfile()
-		performanceManifest, found := os.LookupEnv("PERFORMANCE_PROFILE_MANIFEST_OVERRIDE")
-		if found {
+		profileAlreadyExists := false
+
+		if discovery.Enabled() {
 			var err error
-			performanceProfile, err = externalPerformanceProfile(performanceManifest)
-			Expect(err).ToNot(HaveOccurred(), "Failed overriding performance profile", performanceManifest)
-			klog.Warning("Consuming performance profile from ", performanceManifest)
+			performanceProfile, err = profiles.GetByNodeLabels(
+				map[string]string{
+					fmt.Sprintf("%s/%s", testutils.LabelRole, testutils.RoleWorkerCNF): "",
+				},
+			)
+			Expect(err).ToNot(HaveOccurred(), "Failed finding a performance profile in discovery mode")
+			profileAlreadyExists = true
+			klog.Info("Discovery mode: consuming a deployed performance profile from the cluster")
+		} else {
+			performanceManifest, found := os.LookupEnv("PERFORMANCE_PROFILE_MANIFEST_OVERRIDE")
```

Comment:
I would put this part together with the creation.
Also, even if we are now adding this discovery mode, there's value in having a way to create a test profile and use it in subsequent test runs (this is because applying the profile is time consuming). Even when not in discovery mode, I'd try to see if the profile with the given name exists and in that case not do anything, maybe with a more explicit name (something like testprofile)

Reply:
posted changes that should answer these requested changes. hope the logic is not too complex now.